src/v/kafka/client/sasl_client.cc

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#include "kafka/client/sasl_client.h"
+
+namespace kafka::client {
+
+ss::future<> do_sasl_handshake(shared_broker_t broker, ss::sstring mechanism) {
+    sasl_handshake_request req;
+    req.data.mechanism = std::move(mechanism);
+
+    auto resp = co_await broker->dispatch(req);
+    if (resp.data.error_code != error_code::none) {
+        throw broker_error{broker->id(), resp.data.error_code};
+    }
+}
+
+static ss::future<security::server_first_message> send_scram_client_first(
+  const shared_broker_t& broker,
+  const security::client_first_message& client_first) {
+    sasl_authenticate_request client_first_req;
+    {
+        auto msg = client_first.message();
+        client_first_req.data.auth_bytes = bytes(msg.cbegin(), msg.cend());
+    }
+    auto client_first_resp = co_await broker->dispatch(client_first_req);
+    co_return security::server_first_message(client_first_resp.data.auth_bytes);
+}
+
+static ss::future<security::server_final_message> send_scram_client_final(
+  const shared_broker_t& broker,
+  const security::client_final_message& client_final) {
+    sasl_authenticate_request client_last_req;
+    {
+        auto msg = client_final.message();
```

Comment:
the format of the message() makes sense now. I think we should add a unit test or a ducktape to this to ensure it doesn't break.